cpp/src/arrow/builder.h

```diff
@@ -252,7 +252,7 @@ class ARROW_EXPORT PrimitiveBuilder : public ArrayBuilder {
   ARROW_DEPRECATED("Use Finish instead")
   std::shared_ptr<Buffer> data() const { return data_; }
 
-  const value_type GetValue(int64_t index) const {
+  value_type GetValue(int64_t index) const {
     return reinterpret_cast<const value_type*>(data_->data())[index];
```

Comment:
This will probably fail in CI as long as there is also a const qualifier in `reinterpret_cast<const value_type*>`?

Reply:
Oh, absolutely. Thx.

Reply:
Interstingly, it failed because of ARROW-2831. The compiler doesn't seem to be bothered, but it obviously doesn't make sense to have a const there.

Reply:
Ok, I was confused here. The const is required because data() returns a const pointer. But we can return a non-const value, because we are copying it.